src/irmin-unix/cli.ml

```diff
@@ -535,14 +546,60 @@ let watch =
                   in
                   let* x = x in
                   let* y = y in
-                  let* diff = S.Tree.diff x y in
-                  List.iter pr diff;
-                  Lwt.return_unit)
+                  let* (diff :
+                         (S.path * (S.contents * S.metadata) Irmin.Diff.t) list)
+                      =
+                    S.Tree.diff x y
+                  in
+                  match command with
+                  | h :: t ->
+                      let s =
+                        Fmt.str "%a"
+                          Irmin.Type.(
+                            pp_json
+                              (list
+                                 (pair S.path_t
+                                    (Irmin.Diff.t
+                                       (pair S.contents_t S.metadata_t)))))
+                          diff
+                      in
+                      let make_proc () =
+                        let p =
+                          Lwt_process.open_process_out
+                            (h, Array.of_list (h :: t))
+                        in
+                        proc := Some p;
+                        p
+                      in
+                      let proc =
```

Comment:
can you extract this code in a subfunction? It's a bit hard to review what this is doing exactly here.

Reply:
Sure, I also added some comments to make it easier to follow